src/Map.jl

```diff
@@ -40,6 +59,24 @@ end
 #
 ################################################################################
 
+@doc raw"""
+    identity_map(R::D) where D <: AbstractAlgebra.Set
+
+Return an identity map on the domain $R$.
+
+# Examples
+```jldoctest; setup = :(using AbstractAlgebra)
+julia> hom = identity_map(ZZ)
```

Comment:
So I've used the advanced debugging technique of inserting `println` statements into the Documenter code to learn that this line is causing the problems (or at least it is the first one I am hitting right now).

Documenter first runs this code successfully:
```julia
            c = IOCapture.capture(rethrow = InterruptException) do
                Core.eval(sandbox, ex)
            end
```
where `ex` is equal to the following quoted code:
```
begin
    $(Expr(:softscope, true))
    f = AbstractAlgebra.identity_map(AbstractAlgebra.ZZ)
end
```
But then it proceeds to this, which throws an exception because `c.value` is not defined:
```julia
            Core.eval(sandbox, Expr(:global, Expr(:(=), :ans, QuoteNode(c.value))))
```

But how is that possible? Hrm

Reply:
Just to explain, I know that the access to `c.value` is throwing the exception because I moved it to a separate line; specifically, I inserted this code before the `Core.eval` and after the call to `IOCapture.capture:
```julia
            println("IOCapture.capture complete")
            println("typeof(c) = $(typeof(c))")
            println("c = $c")
            println("c.value = $(c.value)")
```
and got this output:
```
IOCapture.capture complete
typeof(c) = NamedTuple{(:value, :output, :error, :backtrace), Tuple{AbstractAlgebra.Generic.IdentityMap{AbstractAlgebra.Integers{BigInt}}, String, Bool, Vector{Ptr{Nothing}}}}
┌ Error: Doctesting failed
│   exception =
│    UndefRefError: access to undefined reference
│    Stacktrace:
│      [1] show(io::IOBuffer, t::NamedTuple{(:value, :output, :error, :backtrace), Tuple{AbstractAlgebra.Generic.IdentityMap{AbstractAlgebra.Integers{BigInt}}, String, Bool, Vector{Ptr{Nothing}}}})
│        @ Base ./namedtuple.jl:164
...
```
Very, very strange. I'll try and see if I can use something more advanced like Infiltrator to learn more.

Reply:
(P.S. I found it a bit faster with `JULIA_DEBUG=Documenter`) but I don't understand how a named tuple can have undefined references.

Reply:
Infilitrator didn't really help because it could not enter after `c` was created (I guess that's because it looks at all local variables and touching `c` in any way blows up).

The type for the `NamedTuple` also explicitly says that the `value` component has type `AbstractAlgebra.Generic.IdentityMap{AbstractAlgebra.Integers{BigInt}}`. So there shouldn't be a way for it to be undefined? But still a `UndefRefError` is thrown, by `getfield`... Indeed:
```julia
            println("IOCapture.capture complete")
            println("typeof(c) = $(typeof(c))")
            println("nfields(c) = $(nfields(c))")
            println("fieldtype(c,1) = $(fieldtype(typeof(c), 1))")
            println("getfield(c,1) = $(getfield(c, 1))")
            println("c = $c")
            println("c.value = $(c.value)")
```
results in this output:
```
IOCapture.capture complete
typeof(c) = NamedTuple{(:value, :output, :error, :backtrace), Tuple{AbstractAlgebra.Generic.IdentityMap{AbstractAlgebra.Integers{BigInt}}, String, Bool, Vector{Ptr{Nothing}}}}
nfields(c) = 4
fieldtype(c,1) = AbstractAlgebra.Generic.IdentityMap{AbstractAlgebra.Integers{BigInt}}
┌ Error: Doctesting failed
│   exception =
│    UndefRefError: access to undefined reference
...
```